onset_maproom/agronomy.py

```diff
@@ -367,123 +368,93 @@ def soil_plant_water_balance(
     reduced (or actual) evapotranspiration,
     drainage
     """
-    # Start water balance ds with runoff
+    # Runoff
     if runoff is None:
-        runoff = xr.zeros_like(daily_rain).rename("runoff")
-        runoff.attrs = dict(description="Runoff", units="mm")
-    water_balance = xr.Dataset().merge(runoff)
+        runoff = xr.DataArray(np.zeros(1)).squeeze("dim_0")
```

Comment:
Is this different from saying `runoff = 0`? I thought xarray was capable of broadcasting a number to a DataArray when it needs to. (Similarly elsewhere for other constants.)

Reply:
In those cases, it's not a broadcasting matter, but just to be able to apply xarray functions (.i/sel, .rename, merge, .dims, .where) to the variable regardless whether it's a number or already an xarray. Either I do it like this, or I make cases each time they (runoff, kc, et) are used. Or maybe there is another path?
The runoff case is simpler as it's not interfering with the loop so it's xarray nature is only useful to merge it to the dataset output at the end. kc and et are more involved.

Reply:
Any chance to look further into this, @aaron-kaplan ?

Reply:
That makes sense.

I think you can create the 0-d DataArray more succinctly:
```
runoff = xr.DataArray(0.)
```

Reply:
just pushed a commit to simplify creation of constant arrays